webui/react/src/pages/TrialDetails/TrialChart.tsx

```diff
@@ -110,9 +111,15 @@ const TrialChart: React.FC<Props> = ({
   return (
     <Section bodyBorder options={options} title="Metrics">
       <div className={css.base}>
-        {chartData[0].length === 0 ?
-          <Empty description="No data to plot." image={Empty.PRESENTED_IMAGE_SIMPLE} /> :
-          <UPlotChart data={chartData} options={chartOptions} />}
+        {
+          <Spinner className={css.spinner} conditionalRender spinning={!trialId}>
```

Comment:
similar thing, doing the conditional render makes it less obvious what the page is going to be while it is loading.

would be good to have like a skeleton chart or something we could fall back to.